pkg/js/global/scripts.go

```diff
@@ -111,11 +114,16 @@ func initBuiltInFunc(runtime *goja.Runtime) {
 		},
 		Description: "isPortOpen checks if given TCP port is open on host. timeout is optional and defaults to 5 seconds",
 		FuncDecl: func(host string, port string, timeout ...int) (bool, error) {
-			timeoutInSec := 5
+			ctx := context.Background()
 			if len(timeout) > 0 {
-				timeoutInSec = timeout[0]
+				var cancel context.CancelFunc
+				ctx, cancel = context.WithTimeout(ctx, time.Duration(timeout[0])*time.Second)
+				defer cancel()
 			}
-			conn, err := net.DialTimeout("tcp", net.JoinHostPort(host, port), time.Duration(timeoutInSec)*time.Second)
+			if host == "" || port == "" {
+				return false, errkit.New("isPortOpen: host or port is empty")
+			}
```

Comment:
DRY with lines 148-150, could be simplified by turning it into a util function.

Reply:
not sure tbh, this feels simple enough , in go due to its syntax condition with early exit will always require 2 lines of code, we have a helper function in utils `generics` package and could do something like `generics.EqualsAny("",host,port)` but feels like it will overcomplicate things instead of KISS
let me know if you feel otherwise